image/templates/helm/shared/internal/scanner-v4-config-shape.yaml

```diff
@@ -0,0 +1,79 @@
+scannerV4:
```

Comment:
The issue is there will be a period of time where old Scanner and the new Scanner v4 must live side-by-side, so we must make it clear they are different, so we can't just reuse `scanner`. I don't know of a better name at this time, and even the images are called `scanner-v4*`.

Yes, it could be annoying for when/if there is a Scanner v5, but I imagine the Scanners would have to co-exist then, too, and we'd need to make sure they are separately configurable as well

Reply:
So due to backwards compatibility requirements, and the fact that we occasionally want scanner versions to coexist in the same installation, we'll have to keep all the old names. `scanner` for now, but also `scannerV4` after it's replaced, and so on. This might become confusing in the future, but then again I also don't have an alternative proposal.

Reply:
I have been thinking about an alternative here. What about this...

We could -- in addition to the current semantics -- allow `scanner` to be an array of scanner configurations.

Something like:

```
scanner: {...}
``

could be equivalently written as:
```
scanner: [{...}]
```

The existing scanner configuration would be supported as is. Under the hood this could correspond to a new field `version` within the scanner configuration defaulting to "2".

In addition, one could explicitly enable scanner V4 (in addition to V2) by writing:

```
scanner:
  - {"version": "2" /* this could be either present or missing */, ...}
  - {"version": "4", ...}
```

The Helm chart would contain the logic for feeding different scanner configurations into either `_rox.scanner` or `_rox.scannerV4`, depending on the version tag.

Of course, in the future one could extend this in a straight forward manner.

Reply:
Interesting idea @mclasmeier. If we expect to have similar scenarios in the future where we want to support running multiple scanners with similar configuration field this would be a good fit.

For me it appears with a list based approach we'd have to make some sacrifices:
- Validation with OpenAPI scheme becomes more challenging, because list entries will probably not hold exactly the same fields
- We'd have to handle the conflict between the current scanner field changing to a list instead which would mean we need a breaking change in the CRD before scannerv4 is even in tech preview
- The list approach appears more complex to a customer, especially at a point were we go back to a single scanner again

If I understand correctly this is a one time change and we don't expect anything like this happening in mid/long term, so I wonder if it's worth to sacrifice validation and clarity in favor a list based approach.

So maybe `scannerV4` if that's what this feature is called for customers is indeed the right choice for the field name.